cli/flox-rust-sdk/src/models/environment/path_environment.rs

```diff
@@ -483,25 +483,33 @@ impl PathEnvironment {
         replaced = replaced.replace(FLOX_INSTALL_PLACEHOLDER, packages);
 
         // Replace profile
-        let profile = if let Some(ref custom_profile) = customization.profile {
+        let profile = formatdoc! {r#"
+            # common = """
+            #     fortune | lolcat
+            # """
+            # bash = """
+            #     set -o vi
+            # """
+            # zsh = """
+            #     bindkey -v
+            #     bindkey "^R" history-incremental-search-backward
+            # """"#};
+        replaced = replaced.replace(FLOX_PROFILE_PLACEHOLDER, &profile);
+
+        // Replace the hook
+        let default_hook = if let Some(ref custom_script) = customization.script {
```

Comment:
The init customizations where explicitly put in the profile section as some tools do set aliases or provide functions (e.g. venv)

Reply:
Not sure what you mean? Yes users can put customizations in both sections - hook.on-activate for things to be once per activation and profile.* for things to run in all (sub)shell activations. Can you clarify the suggestion? Thanks.

Reply:
@ysndr I now understand what you mean in light of Zach's comments. Including a `venv` example would be useful. Maybe this:
```
[hook]
on-activate = """
    venv_dir=$(mktemp -d)
    export venv_dir
    python -m venv $venv_dir
"""

[profile]
bash = """
    source $venv_dir/bin/activate
"""
zsh = """
    source $venv_dir/bin/activate
"""
fish = """
    source $venv_dir/bin/activate.fish
"""
tcsh = """
    source $venv_dir/bin/activate.csh
"""
```

... definitely looking for product & marketing input on this.

Reply:
That's why I left the comment. I think the init customizations still belong in the `profile` section for this reason, and changing which script they show up in feels orthogonal to what this PR is trying to accomplish.

Reply:
It also reduces the scope of the PR a little bit, which is nice.

Reply:
I think eventually what we want is the ability to put init customizations in either script, but for now let's leave it how it is.

Reply:
> I think the init customizations still belong in the profile section for this reason ...

I genuinely don't know what you mean by that, because if you performed the `python -m venv` in any of the `[profile]` sections then that would be repeated with every new subshell.

???

Reply:
First, what do you mean by "subshell", is it intended behavior that

```
$ flox activate -d one
[one] $ flox activate -d two
[two one] $
```

executes the profile scripts for `one` twice contrary to on-activate which will only run once in any subshell hierarchy?
At least that's my interpretation (though last time i checked this didn't hold).

Reply:
Yes that is the intended behaviour. The good way to look at this is to think about aliases which are not passed down by way of the environment. In order for aliases to work in all subshells it's essential that profile scripts for all active environments be sourced for all subshell activations.

In reality this is most important for the default environment, which people will activate by way of "in place" activations invoked from their `.{bashrc,zshrc,zshenv}` files. In that regard if I have an alias I want it to work everywhere, regardless whether it's a "login" shell, a pane of a `tmux` invocation, or the result of activating another flox environment.

Reply:
I pushed 8815d18f9f1809621e22cb909e3e7adf0179b3ac which updates the template as noted above, plus a few more tweaks. I'm increasingly of the opinion that we should put a pin in the conversation here and continue efforts to perfect the template in a separate PR, so I'll go ahead and resolve this conversation and add another task to the follow-up issue #1343.

Reply:
I get the feeling that `profile` is really supposed to be a second tier thing and that basically anything you want done in terms of a hook should go into `on-activate`.
To me `profile` behaves exclusively annoying except when you want to do shell modifications ^^


Wrt. python activation:
0) do we care about the venv `deactivate` function/alias at all in the first place?
1) iiuc all these python tools work by setting `VIRTUAL_ENV=...`, don't layer at all and also set `PS1` all of which might have possible annoying side effects in a `profile` script but depending on how things are ordered.
2) I figure long term flox might be able to implement setting up `VIRTUAL_ENV` in a floxy way through e.g. etc-profiles

Reply:
In other conversations we've arrived at the idea that developer environments can only be activated as the top layer, and that lower-layer environments are relegated to "runtime" status when that happens. So with that this may well be a simple replaying of environment variable setting/unsetting rather than a complex merging of state, but you're absolutely right that etc-profiles provides the means by which to "do the needful" if we find that is necessary!

Reply:
Will look to reproduce with a test, then will fix.  :+1: